WordPress/WordPressShare/PostStatusPickerViewController.swift

```diff
@@ -0,0 +1,98 @@
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupView()
+        setupTableView()
+        setupNoResultsView()
+        loadStatuses()
+    }
+    
+    
+    // MARK: - UITableView Methods
+    override func numberOfSectionsInTableView(tableView: UITableView) -> Int {
+        return 1
+    }
+    
+    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return statuses.count
+    }
+    
+    override func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
+        return rowHeight
+    }
+    
+    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        var cell = tableView.dequeueReusableCellWithIdentifier(reuseIdentifier)
+        if cell == nil {
+            cell = WPTableViewCell(style: .Subtitle, reuseIdentifier: reuseIdentifier)
+            WPStyleGuide.Share.configureBlogTableViewCell(cell!)
+        }
+        
+        let status = sortedStatuses[indexPath.row].0
+        configureCell(cell!, status: status)
+        
+        return cell!
+    }
+    
+    override func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
+        let status = sortedStatuses[indexPath.row].0
+        let description = statuses[status]!
+        onChange?(postStatus: status, description: description)
+        navigationController?.popViewControllerAnimated(true)
+    }
+    
+    
+    // MARK: - Setup Helpers
+    private func setupView() {
+        title = NSLocalizedString("Post Status", comment: "Title for the Post Status Picker")
+    }
+    
+    private func setupTableView() {
+        // Blur!
+        let blurEffect = UIBlurEffect(style: .Light)
+        tableView.backgroundColor = UIColor.clearColor()
+        tableView.backgroundView = UIVisualEffectView(effect: blurEffect)
+        tableView.separatorEffect = UIVibrancyEffect(forBlurEffect: blurEffect)
+        
+        // Fix: Hide the cellSeparators, when the table is empty
+        tableView.tableFooterView = UIView()
+    }
+    
+    private func setupNoResultsView() {
+        noResultsView = WPNoResultsView()
+        tableView.addSubview(noResultsView)
+    }
+    
+    
+    // MARK: - Private Helpers
+    private func configureCell(cell: UITableViewCell, status: String) {
+        // Status' Details
+        let statusDescription = statuses[status]
+        cell.textLabel?.text = statusDescription
+    }
+    
+    private func loadStatuses() {
+        sortedStatuses = statuses.sort({ $0.1 < $1.1 })
+    }
+    
+    // MARK: Typealiases
+    typealias PickerHandler = (postStatus: String, description: String) -> Void
+    
+    // MARK: - Public Properties
+    var onChange                : PickerHandler?
+    
+    // MARK: - Private Properties
+    var statuses: [String: String]!
+    private var sortedStatuses: [(String, String)]!
```

Comment:
Instead of so many (implicitly unwrapped) optionals, why not initialize `PostStatusPickerViewController` with a list of `statuses`, and store them already sorted.

Also, it's not clear what's the key and what's the value unless you look at what the code does with them. This should have some documentation.